integration-test/http/http_system_test.c

```diff
@@ -0,0 +1,383 @@
+#else
+    #define IOT_CORE_ALPN_PROTOCOL_NAME_LENGTH    ( sizeof( IOT_CORE_ALPN_PROTOCOL_NAME ) - 1 )
+#endif
+
+/* Check that size of the user buffer is defined. */
+#ifndef USER_BUFFER_LENGTH
+    #define USER_BUFFER_LENGTH    ( 2048 )
+#endif
+
+/**
+ * @brief Length of HTTP server host name.
+ */
+#define SERVER_HOST_LENGTH     ( ( uint16_t ) ( sizeof( SERVER_HOST ) - 1 ) )
+
+/**
+ * @brief Length of the request body.
+ */
+#define REQUEST_BODY_LENGTH    ( sizeof( REQUEST_BODY ) - 1 )
+
+/**
+ * @brief Represents the OpenSSL context used for TLS session with the broker
+ * for tests.
+ */
+static NetworkContext_t networkContext;
+
+/**
+ * @brief The transport layer interface used by the HTTP Client library.
+ */
+static TransportInterface_t transportInterface;
+
+/**
+ * @brief Represents the hostname and port of the broker.
+ */
+static ServerInfo_t serverInfo;
+
+/**
+ * @brief TLS credentials needed to connect to the broker.
+ */
+static OpensslCredentials_t opensslCredentials;
+
+/**
+ * @brief Shared buffer used in the tests for storing HTTP request headers and
+ * HTTP response headers and body.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Connect to HTTP server with reconnection retries.
+ *
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ *
+ */
+static void connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path, then
+ * print the response received from the server.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ */
+static void sendHttpRequestWithBackoffRetries( const TransportInterface_t * pTransportInterface,
+                                               const char * pMethod,
+                                               const char * pPath );
+
+/*-----------------------------------------------------------*/
+
+static void connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
+{
+    /* Status returned by the retry utilities. */
+    RetryUtilsStatus_t retryUtilsStatus = RetryUtilsSuccess;
+    /* Struct containing the next backoff time. */
+    RetryUtilsParams_t reconnectParams;
+    /* Status returned by OpenSSL transport implementation. */
+    OpensslStatus_t opensslStatus;
+
+    /* Reset or initialize file-scoped global variables. */
+    ( void ) memset( &opensslCredentials, 0, sizeof( opensslCredentials ) );
+    opensslCredentials.pRootCaPath = ROOT_CA_CERT_PATH;
+    opensslCredentials.pClientCertPath = CLIENT_CERT_PATH;
```

Comment:
Let's delete lines 157-160 because we are not doing the integration tests against AWS IoT. AWS IoT only supports the POST method.

Reply:
Removed them in the most recent commit :)